src/overlay.py

```diff
@@ -63,9 +65,22 @@ def add_attribute(
         attribute_format_as: type,
     ) -> None:
         """Add an attribute to the overlay. The attribute can be formatted as int, float, or str."""
-        self.line_order.append(
-            (obj, attribute_name, attribute_display_name, attribute_format_as)
-        )
+
+        if (
+            hasattr(obj, attribute_name)
+            or isinstance(obj, LinearPhysicalObject)
+            and hasattr(obj.kinematic, attribute_name)
```

Comment:
Do i get it right, that here obj is checked on having a kinematic, but the treatment only happens in some private method? this is difficult to oversee and understand. One more reason to simplify to a non-extrawurscht-for-kinematic-solution